Cargo.toml

```
@@ -18,8 +18,11 @@ lazy_static = "1.2"
 rand = "0.6"
 
 [features]
-no_fail = []
+failpoints = []
```

Perhaps make `failpoints` as the default feature? Dependencies not wanting failpoints could still opt-out with `default-features = false`.

```suggestion
default = ["failpoints"]
failpoints = []
```

---

I understand the sentiment behind this suggestion, but unfortunately that will nullify this PR. There is no way for cargo to disable default features across transitive dependencies (by design). It does provide a way to enable them though, via the `crate/feature` syntax.

This library features should be handled by chaining upward the opt-in switch, and leaving the final decision to the top consumer.

I acknowledge this is a breaking change, but at the same time I do believe failpoints are an instrumentation feature mostly meant for testing/debugging mode (i.e. not normally present in release-mode binary)

---

How about naming it activation?

/cc @brson @siddontang @overvenus PTAL

---

@kennytm Do you have any suggestions?